test/unit/oracle/PCVOracle.t.sol

```diff
@@ -71,53 +79,74 @@ contract PCVOracleUnitTest is Test {
         core.grantLocker(address(pcvOracle));
         core.grantLocker(address(deposit1));
         core.grantLocker(address(deposit2));
+        core.setPCVOracle(IPCVOracle(address(pcvOracle)));
+
+        core.createRole(VoltRoles.PCV_DEPOSIT, VoltRoles.GOVERNOR);
+        core.grantRole(VoltRoles.PCV_DEPOSIT, address(deposit1));
+        core.grantRole(VoltRoles.PCV_DEPOSIT, address(deposit2));
+
+        oracle1.setValues(1e18, true);
+        oracle2.setValues(1e18, true);
+
+        // add venues
+        address[] memory venues = new address[](2);
+        venues[0] = address(deposit1);
+        venues[1] = address(deposit2);
+        address[] memory oracles = new address[](2);
+        oracles[0] = address(oracle1);
+        oracles[1] = address(oracle2);
+        pcvOracle.addVenues(venues, oracles);
+
         vm.stopPrank();
     }
 
     function testSetup() public {
-        assertEq(pcvOracle.getVenues().length, 0);
+        assertEq(pcvOracle.getVenues().length, 2);
+        assertEq(pcvOracle.getNumVenues(), 2);
         uint256 totalPcv = pcvOracle.getTotalPcv();
         assertEq(totalPcv, 0);
-    }
+        assertEq(pcvOracle.lastRecordedTotalPcv(), 0);
+
+        {
+            (uint128 balance, int128 profit) = pcvOracle.venueRecord(
+                address(deposit1)
+            );
+            assertEq(balance, 0);
+            assertEq(profit, 0);
+        }
+        {
+            (uint128 balance, int128 profit) = pcvOracle.venueRecord(
+                address(deposit2)
+            );
+            assertEq(balance, 0);
+            assertEq(profit, 0);
+        }
 
-    function testGetTotalPcvFailsWhileEntered() public {
-        vm.prank(address(deposit2));
-        lock.lock(1);
+        assertEq(pcvOracle.venueToOracle(address(deposit1)), address(oracle1));
+        assertEq(pcvOracle.venueToOracle(address(deposit2)), address(oracle2));
+    }
 
-        vm.expectRevert("PCVOracle: cannot read while entered");
-        pcvOracle.getTotalPcv();
+    function testDecimalNormalization() public {
+        oracle1.setValues(1e12, true); /// only add 12 decimals on deposit, this means 6 are truncated from 18
 
-        vm.prank(address(deposit2));
-        lock.lock(2);
+        token1.mint(address(deposit1), 100e18);
+        entry.deposit(address(deposit1));
 
-        vm.expectRevert("PCVOracle: cannot read while entered");
-        pcvOracle.getTotalPcv();
+        assertEq(pcvOracle.lastRecordedPCV(address(deposit1)), 100e12); /// with scale up
+        assertEq(pcvOracle.lastRecordedPCVRaw(address(deposit1)), 100); /// remove scale up
```

Comment:
That seems messed up, is 100e18 tokens are minted to the deposit, last balance should be 100e18 and lastRecordedPCV should be whatever the oracle value is